### DISCUSSION TITLE
Aspect ratio/scaling question

### DISCUSSION THREAD

**Discussion by Pirateguybrush** [hansschmucker/NVStreamer1080#13] [Q&A]
Hello!

I'm trying to replicate this functionality within DisplayFusion, as I already use that tool to manage my displays.

Any chance you could explain how this software functions?

I can get it to automatically change to 1280x800 when nvstreamer.exe starts, and return when it ends.  However the scaling is off, and moonlight displays the screen in a corner of the Steam Deck's screen, rather than fullscreen.

If I change the scaling or resolution settings in the nvidia control panel though, it fixes it.   This suggests there's an additional setting that needs to be triggered to get the scaling to work properly, perhaps?  I feel like I'm close, and would love it if you'd point me in the right direction to get this working.

Your tool is great and works perfectly - I'm just keen to minimise the number of active applications on my PC, and DisplayFusion is already doing a fair bit for me.  I'm sure it could be tweaked to handle this too.

**Comment by hansschmucker** [comment]
Hi @Pirateguybrush
There's nothing really special going on here... I don't know DisplayFusion and how quickly it reacts, but my guess is that it's changing the resolution too late.
Other than that it's really just 2 simple WinAPI calls using user32.dll:

`
            var disp = new DispSet();
            EnumDisplaySettings(null, -1, ref disp);
            disp.width = w;
            disp.height = h;
            disp.frequency = r;
            ChangeDisplaySettings(ref disp, 1);
`
Sorry I can't be of any more help, but that's really all I'm doing here ^^
